xml/hwbp_registers.xml

```diff
@@ -980,8 +980,12 @@
     <register name="Instruction Count" short="icount" address="0x7a1">
         This register is accessible as \RcsrTdataOne when \FcsrTdataOneType is 3.
 
-        This trigger matches on every instruction completed or trap taken from a
-        privilege mode where the trigger is enabled.
+        \begin{steps}{This trigger matches when:}
+            \item An instruction is fetched in a privilege mode where the
+                trigger is enabled and then retires.
+                This explicitly includes {\em x}\/RET instructions.
```

Comment:
I find the wording awkward (particularly the "and then retires" being split from "fetched" by the intervening phrase).  Is there a way to say that it retires in an enabled mode (leaving out fetch) and then the second sentence says that in this context xRET is considered to retire in the mode it's coming from (or fetched in) rather than the mode it's going to?  Or even "An instruction retires after having been fetched in a privilege mode where the trigger is enabled" would be better in my view.

The content is good.  Just wordsmithing.

Reply:
I like your second wording, because it makes the line about xRET be simply a clarification and not a special case. If there are/will be other instructions that change the privilege mode we want them to also be included.